deeplearning4j-nn/src/main/java/org/deeplearning4j/nn/conf/ocnn/OCNNOutputLayer.java

```diff
@@ -0,0 +1,110 @@
+package org.deeplearning4j.nn.conf.ocnn;
+
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+import lombok.NoArgsConstructor;
+import lombok.ToString;
+import org.deeplearning4j.nn.api.Layer;
+import org.deeplearning4j.nn.api.ParamInitializer;
+import org.deeplearning4j.nn.conf.NeuralNetConfiguration;
+import org.deeplearning4j.nn.conf.layers.BaseOutputLayer;
+import org.deeplearning4j.nn.conf.layers.LayerValidation;
+import org.deeplearning4j.nn.layers.ocnn.OCNNParamInitializer;
+import org.deeplearning4j.optimize.api.TrainingListener;
+import org.nd4j.linalg.api.ndarray.INDArray;
+import org.nd4j.shade.jackson.annotation.JsonCreator;
+import org.nd4j.shade.jackson.annotation.JsonIgnoreProperties;
+import org.nd4j.shade.jackson.annotation.JsonProperty;
+
+import java.util.Collection;
+import java.util.Map;
+
+/**
+ * An implementation of one class neural networks from:
+ * https://arxiv.org/pdf/1707.02131.pdf
```

Comment:
Can we use this one? :) I was super confused for a while. I mean, the other paper is interesting as well.

https://arxiv.org/pdf/1802.06360.pdf

Reply:
done.